share/shwap/range_namespace_data.go

```diff
@@ -375,26 +390,23 @@ func generateSharesProofs(
 	namespace libshare.Namespace,
 	rowShares []libshare.Share,
 ) (*nmt.Proof, error) {
-	tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(size), uint(row))
-
-	for i, shr := range rowShares {
-		if err := tree.Push(shr.ToBytes()); err != nil {
-			return nil, fmt.Errorf("failed to build tree at share index %d (row %d): %w", i, row, err)
-		}
-	}
-
-	root, err := tree.Root()
+	// Create RowNamespaceData for the entire row first
+	rowData, err := RowNamespaceDataFromShares(rowShares, namespace, row)
```

Comment:
wait why are we doing this here? This feels excessive just to check for NS.

Reply:
yes, it doesn't make sense to do this refactor. reverted it back.
in terms of the difference, the RowNamespaceDataFromShares derives the prove range and also does implicit handling of the non-inclusion proof.